spec/models/rating_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe Rating, type: :model do
+  let(:rating) { Rating.create }
+  let(:user) { FactoryGirl.create(:user) }
+
+  context 'when change value' do
+    context '#increase' do
+      it 'increase rating by 1' do
+        expect { rating.increase_by(user: user) }.to change { rating.value }.by(1)
+      end
+
+      it 'increase rating only once' do
+        rating.increase_by(user: user)
+        expect { rating.increase_by(user: user) }.to raise_error(ActiveRecord::RecordInvalid)
+      end
+
+      it 'adds user to upvoters' do
+        rating.increase_by(user: user)
+        expect(rating.reload.upvoters).to include(user)
+      end
+
+      it 'not adds user to downvoters' do
+        rating.increase_by(user: user)
+        expect(rating.reload.downvoters).to_not include(user)
+      end
+
+      it 'change rated status' do
+        expect(rating.increased_by?(user)).to be_falsey
+        rating.increase_by(user: user)
+        expect(rating.reload.increased_by?(user)).to be_truthy
+      end
+
+      it 'removes user from decreased' do
```

Comment:
Looks like you're not testing the `value` change when previously-downvoted post gets upvoted. I'd say you should, since those manual counter caches can be quite fragile in my experience.